packages/core/helpers/router-method-factory.ts

```diff
@@ -1,28 +1,31 @@
 import { HttpServer } from '@nestjs/common';
 import { RequestMethod } from '@nestjs/common/enums/request-method.enum';
 
+/**
+ * Ensures (via satisfies) there's a mapping between the RequestMethod enum and the HttpServer interface.
+ * @note This is a compile-time check, so if the interface changes, the compiler will complain.
+ * This is to resolve a case where a new RequestMethod is added, but the RouterMethodFactory is not updated.
+ */
+const RequestMethodMap = {
+  [RequestMethod.GET]: 'get',
+  [RequestMethod.POST]: 'post',
+  [RequestMethod.PUT]: 'put',
+  [RequestMethod.DELETE]: 'delete',
+  [RequestMethod.PATCH]: 'patch',
+  [RequestMethod.ALL]: 'all',
+  [RequestMethod.OPTIONS]: 'options',
+  [RequestMethod.HEAD]: 'head',
+  [RequestMethod.SEARCH]: 'search',
+} as const satisfies Record<RequestMethod, keyof HttpServer>;
+
 export class RouterMethodFactory {
   public get(target: HttpServer, requestMethod: RequestMethod): Function {
-    switch (requestMethod) {
-      case RequestMethod.POST:
-        return target.post;
-      case RequestMethod.ALL:
-        return target.all;
-      case RequestMethod.DELETE:
-        return target.delete;
-      case RequestMethod.PUT:
-        return target.put;
-      case RequestMethod.PATCH:
-        return target.patch;
-      case RequestMethod.OPTIONS:
-        return target.options;
-      case RequestMethod.HEAD:
-        return target.head;
-      case RequestMethod.GET:
-        return target.get;
-      default: {
-        return target.use;
-      }
+    const methodName = RequestMethodMap[requestMethod];
+    const method = target[methodName];
+    if (!method) {
+      // There should probably be a warning message in this case
```

Comment:
```suggestion
      // There should probably be a warning message in this case
```
Using .use by default is intentional (nothing wrong with that)

Reply:
hence the "probably", the issue is that silently using `.use` because there was no other method found is the cause of this whole issue. it will hide any other controller method the follows.

Reply:
removed comment